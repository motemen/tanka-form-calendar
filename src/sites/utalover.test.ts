import うたらば from "./utalover";

describe("utalover", () => {
  it("crawl", async () => {
    const events = await うたらば.crawl();
    expect(events).toMatchSnapshot();

    expect(new Set(events.map((r) => うたらば.eventDetail(r)))).toEqual(
      new Set([
        { title: "フリーペーパーうたらば『飲み物』", url: "https://www.utalover.com/" },
        { title: "月刊うたらば『会』", url: "https://www.utalover.com/" },
      ])
    );
  });
});
