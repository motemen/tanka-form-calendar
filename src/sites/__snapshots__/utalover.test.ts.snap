// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utalover crawl 1`] = `
Array [
  Object {
    "date": Array [
      2022,
      7,
      2,
    ],
    "detail": Object {
      "collection": "月刊うたらば",
      "theme": "会",
    },
    "key": "9fda14704cbec22d3dac46a1273574a3",
  },
  Object {
    "date": Array [
      2022,
      7,
      23,
    ],
    "detail": Object {
      "collection": "フリーペーパーうたらば",
      "theme": "飲み物",
    },
    "key": "d4c4ed1eec3ed328f9d5db39cb17f9ee",
  },
]
`;
